airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcCtidInitializer.java

```diff
@@ -203,16 +207,24 @@ public static List<AutoCloseableIterator<AirbyteMessage>> cdcCtidIteratorsCombin
             "Postgres database is undergoing a full vacuum - cannot proceed with the sync. Please sync again when the vacuum is finished.");
       }
 
-      finalListOfStreamsToBeSyncedViaCtid = ctidStreams.streamsForCtidSync();
+      // Any stream currently undergoing full vacuum should not be synced via CTID as it is not a stable
+      // cursor. In practice, this will never happen
+      // during a sync as a full vacuum in Postgres locks the entire databa
+      finalListOfStreamsToBeSyncedViaCtid.addAll(ctidStreams.streamsForCtidSync());
```

Comment:
could line 213 - 228 together be replaced by the following single line?

```
 finalListOfStreamsToBeSyncedViaCtid.addAll(finalListOfStreamsToBeSyncedViaCtid.stream()
            .filter(stream -> !fileNodeHandler.getFailedToQuery().contains(
                new AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace())))
            .collect(Collectors.toList()));
``

Reply:
Done!